docs/tutorials/auth.md

```diff
@@ -245,3 +245,156 @@ Access to path `/` would be granted to everyone in `example.org`, however path `
+1. [Add the OAuth2 grant filters to routes.](#add-filters-to-your-routes)
+
+#### Configure OAuth2 credentials
+Before you start, you need to register your application with the OAuth2 provider.
+If your provider asks you for the callback URL, provide the URL that you set 
+as the `-oauth2-callback-path` parameter. If you did not provide a value, use the default
+route : `/.well-known/oauth2-callback`. 
+
+Skipper must be configured with the following credentials and secrets:
+1. OAuth2 client ID for authenticating with the OAuth2 provider.
+1. OAuth2 client secret for authenticating with the OAuth2 provider.
+1. Cookie encryption secret for encrypting and decrypting token cookies.
+
+You can load all of theses secrets from separate files, in which case they get automatically
+reloaded to support secret rotation. You can provide the paths to the files containing each 
+secret as follows:
+
+```sh
+skipper -oauth2-client-id-file=/path/to/client_id \
+    -oauth2-client-secret-file=/path/to/client_secret \
+    -oauth2-secret-file=/path/to/cookie_encryption_secret \
+    -credentials-paths=/path/to/ \
+    -credentials-update-interval=30s
+```
+
+You must set the `-credentials-paths` argument to the directory containing the secrets. You 
+can modify the secret update interval using the `-credentials-update-interval` argument. In
+example above, the interval is configured to reload the secrets from the files every 30
+seconds.
+
+If you prefer, you can provide the client ID and secret values directly as arguments to
+Skipper instead of loading them from files. In that case, call Skipper with:
+
+```sh
+skipper -oauth2-client-id=<CLIENT_ID> -oauth2-client-secret=<CLIENT_SECRET>
+```
+
+#### Configure the grant filters
+
+The grant filters need to be enabled and configured with your OAuth2 provider's
+authorization, token, and tokeninfo endpoints. This can be achieved by providing Skipper
+with the following arguments:
+
+```sh
+skipper -enable-oauth2-grant-flow \
+    -oauth2-auth-url=<OAUTH2_AUTHORIZE_ENDPOINT> \
+    -oauth2-token-url=<OAUTH2_TOKEN_ENDPOINT> \
+    -oauth2-tokeninfo-url=<OAUTH2_TOKENINFO_ENDPOINT> \
+    -oauth2-callback-path=/oauth/callback
+```
+
+You can configure the `oauthGrant()` filter further for your needs. See the
+[oauthGrant](../reference/filters.md#oauthGrant) filter reference for more details.
+
+#### Add filters to your routes
+
+You can protect any number of routes with the `oauthGrant()` filter. Unauthenticated users
+will be refused access and redirected to log in.
+
+Skipper will automatically add a callback route for you with the `grantCallback` filter registered 
+on it. The path for this route can be configured with the `-oauth2-callback-path` parameter.
+If the parameter is not given, it will be `/.well-known/oauth2-callback`
+
+```
+foo:
+    Path("/foo")
+    -> oauthGrant()
+    -> "http://localhost:9090";
+
+bar: 
+    Path("/bar")
```

Comment:
This should be either the default or `/oauth/callback` as defined above in the example how to [start skipper](https://github.com/zalando/skipper/pull/1609/files#diff-f3a465ccf6b9f785970a20701c0841b87572018455dd4e734f20d539d7b5d517R341).

Reply:
Wait, this isn't the callback route, it's just another route to showcase the `oauthGrant` filter. This part of the documentation had an additional route before to show how to register the `grantCallback` route but I removed that in commit e0926923f51de2306674a0b0b56fbd9f96997bef. I guess I could remove one of the two routes though? The second one doesn't really add any value...

Reply:
I agree, remove it

Reply:
Removed in 6825909